pkg/controller/group/group_controller.go

```diff
@@ -106,10 +125,27 @@ func (c *Controller) reconcile(key string) error {
 		return err
 	}
 	if group.ObjectMeta.DeletionTimestamp.IsZero() {
+		var g *iam1alpha2.Group
 		if !sliceutil.HasString(group.Finalizers, finalizer) {
-			group.ObjectMeta.Finalizers = append(group.ObjectMeta.Finalizers, finalizer)
+			g = group.DeepCopy()
+			g.ObjectMeta.Finalizers = append(g.ObjectMeta.Finalizers, finalizer)
+		}
 
-			if group, err = c.ksClient.IamV1alpha2().Groups().Update(group); err != nil {
+		if c.multiClusterEnabled {
+			// Ensure not controlled by Kubefed
+			if group.Labels == nil || group.Labels[constants.KubefedManagedLabel] != "false" {
+				if g == nil {
+					g = group.DeepCopy()
+				}
+				if g.Labels == nil {
+					g.Labels = make(map[string]string, 0)
+				}
+				g.Labels[constants.KubefedManagedLabel] = "false"
```

Comment:
So what if the group doesn't have a `KubefedManagedLabel`, we are going to add a label to it.  I think we'd better do nothing If the group has no label `KubefedManagedLabel`. The L135 
```
    if group.Labels != nil && group.Labels[constants.KubefedManagedLabel] == "true" {
        group.Labels[constants.KubefedManagedLabel] = "false"
    }
```

Reply:
Group will be labeled with KubefedManagedLabel by Kubefed when multi-cluster enabled.  We must prevent Kubefed change Group on the Host cluster by the label.

Reply:
yes, but what if the group without KubefedManagedLabel, why do we still add a KubefedManagedLabel=false to it?

Reply:
This Label will be added only when multi-cluster enabled and on the Master cluster, so it must exist. Immediately or to be updated by Kubefed when FedeatedGroup created.